README.md

```diff
@@ -50,19 +50,56 @@ Commands:
+### Process
 
-Please install pre-commit, once installed the file validations will automatically
-run on every commit. Alternatively you can manually execute the validations by running
-`pre-commit run --all-files`.
+Before making a commit, you should syntactically validate your code and configurations with pre-commit.
 
+You can set up pre-commit hooks to automatically be run before every commit by running: `pre-commit install`.
+Alternatively, you can manually execute the validations by running `pre-commit run --all-files`.
+
+#### Tests
+
+JCCLI's test suite consists of unit tests and integration tests. The integration tests depend on the environment
+variable `JC_API_KEY`, which should be a Jumpcloud API key corresponding to a blank Jumpcloud instance which can be used
+for testing purposes.
+
+We use [Travis-CI](https://travis-ci.org/) to automate our testing. This repo's Travis configuration is set up to run
+the unit test suite (in `unit_tests/`) on every pull request and push, and to run the integration test suite (in
+`integration_tests/`) only on a push.
+
+### GitHub
+
+Contributors are requested to use the following process (in the examples, we'll suppose that a user named `john-smith`
+wants to fix some typos in the documentation):
+
+1. Make a fork of JCCLI. E.g. `Sage-Bionetworks:jccli` &rarr; `john-smith:jccli`
+2. Make a branch off of `master` named after a feature or issue. E.g. `john-smith:jccli/master` &rarr;
+`john-smith:jccli/fix-typo-in-docs`
+3. Make commits to that branch (in this example, `john-smith:jccli/fix-typo-in-docs`). When pushed to GitHub, they
+should trigger Travis to run unit tests and integration tests. For the integration tests to pass, contributors need to
+make sure that the `JC_API_KEY` environmental variable in their Travis CI environment is set to a
+[Jumpcloud API Key](https://jumpcloud.com/demo) &mdash;specifically, one corresponding to a "blank" Jumpcloud instance.
+4. Make a pull request from the feature/issue branch on the fork (e.g. `john-smith:jccli/fix-typo-in-docs`) to
+`Sage-Bionetworks:jccli/master`.
+5. Wait for maintainers to review code and approve the pull request.
+
+Maintainers should use the following process for reviewing and approving outside pull requests:
+
+1. Examine proposed changes on GitHub. Pay special attention to hidden environment variables (as of write time,
+`JC_API_KEY`) and make sure nothing in the changes could expose them or use them for unintended purposes. *If in doubt,
+DO NOT proceed to the next step*.
+2. Incorporate the changes into a new branch in our repo, e.g. create a branch `Sage-Bionetworks:jccli/fix-typo-in-docs`
+and manually pull in the changes from `john-smith:jccli/fix-typo-in-docs`. Push them to remote to trigger a Travis CI
+build. Make sure that the `integration-test` job ran and passed successfully.
+3. Approve the pull request and drop the feature branch made for testing purposes
```

Comment:
does drop mean delete?  

also isn't there a final step to merge `Bionetworks:jccli/fix-typo-in-docs` into `Bionetworks:jccli/master` ?

Reply:
If we have tested it on a new branch in our fork (the original repo, I mean), can't we just approve the PR directly from them?

I.e.:

1. They make a PR: `joe-schmoe:jccli/fix-typo-in-docs` -> `Sage-Bionetworks:jccli/master`
2. We read through the proposal, and then pull their changes from `joe-schmoe:jccli/fix-typo-in-docs` into `Sage-Bionetworks:jccli/fix-typo-in-docs`
3. All tests pass, and then we just approve the merge directly from `joe-schmoe:jccli/fix-typo-in-docs` -> `Sage-Bionetworks:jccli/master`.

Sound good? Or do we absolutely have to merge from `Sage-Bionetworks:jccli/fix-typo-in-docs` -> `Sage-Bionetworks:jccli/master`? My thinking is if we do it the way I just said, we are mostly allowing people to do pull requests in a more normal/direct way.